src/scripts/createForm.js

```diff
@@ -0,0 +1,125 @@
+import { createElement } from './utils';
+import { addRow } from './tableActions';
+import { pushNotification } from './createNotification';
+
+export function createForm(table) {
+  const form = document.createElement('form');
+
+  const selectorOptions = [
+    'Tokyo',
+    'Singapore',
+    'London',
+    'New York',
+    'Edinburgh',
+    'San Francisco',
+  ];
+
+  form.append(
+    createInput('Name:', 'name', 'text'),
+    createInput('Position:', 'position', 'text'),
+    createSelector('Office', 'office', selectorOptions),
+    createInput('Age:', 'age', 'number'),
+    createInput('Salary:', 'salary', 'number'),
+    createSubmitButton(table, form, 'Save to table'),
+  );
+
+  form.classList.add('new-employee-form');
+  document.body.append(form);
+}
+
+function createSelector(labelText, selectorName, selectorOptions) {
+  const label = createElement('label', labelText);
+  const selector = document.createElement('select');
+
+  selectorOptions.forEach((optionText) => {
+    selector.append(createElement('option', optionText));
+  });
+
+  selector.name = selectorName;
+  selector.dataset.qa = selectorName;
+
+  label.append(selector);
+
+  return label;
+}
+
+function createInput(labelText, inputName, inputType = 'text') {
+  const input = document.createElement('input');
+  const label = createElement('label', labelText);
+
+  input.type = inputType;
+  input.name = inputName;
+  input.autocomplete = false;
+  input.dataset.qa = inputName;
+
+  label.append(input);
+
+  return label;
+}
+
+function createSubmitButton(table, form, buttonText) {
+  const submitButton = createElement('button', buttonText);
+
+  submitButton.addEventListener('click', (e) => {
+    e.preventDefault();
+
+    const data = new FormData(form);
+
+    if (data.get('name').length < 4) {
+      return pushNotification(
+        10,
+        10,
+        'Error',
+        'The name is too short.',
+        'error',
+      );
+    }
+
+    if (data.get('position').length < 4) {
+      return pushNotification(
+        10,
+        10,
+        'Error',
+        'The position is too short.',
+        'error',
+      );
+    }
+
+    if (data.get('age') < 18) {
+      return pushNotification(
+        10,
+        10,
+        'Error',
+        'The age is too small.',
+        'error',
+      );
+    }
+
+    if (data.get('age') > 90) {
+      return pushNotification(10, 10, 'Error', 'The age is too big.', 'error');
+    }
+
+    if (data.get('salary') === '') {
```

Comment:
The check for salary being an empty string should also consider if the salary is a valid number. Consider using `isNaN(data.get('salary'))` to ensure the input is a valid number.